Mtl h dgpu

mkopec opened the pull request.

crawfxrd committed:
drivers/i2c/tas5825m: Use I2C instead of SMBus

The latest Clevo boards connect the TAS5825M to one of the I2C
connections instead of the SMBus connection. The I2C ops are compatible
with SMBus, so always use them.

Change-Id: I5152db647094acf473cc798970dd9d97543df4d7
Signed-off-by: Tim Crawford <tcrawford@system76.com>

mkopec committed:
mb/clevo/mtl-h/variants/dgpu: disable dGPU

Change-Id: I37bc576df58dcd30e132f2b43a358136da6f42cc
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/ramstage.c: ensure PTT is disabled

Change-Id: If35a9ec0dfaae4e8aadb53d623ed31420763a20f
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/variants/dgpu: fix TP_ATTN# pin

Change-Id: Id9b5d13f7f25f7f76adbf6b240fee16ac018d753
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/variants/dgpu: uncomment unused variable

Change-Id: I93e401847c3bf24fb19caafa36b1e3403ebec424
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs: add novacustom mtl dgpu configs

Change-Id: I3a3439cb173455b4967ef8265943efda049119aa
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/variants/dgpu: reenable dGPU

Change-Id: I583d818c6980b089487251e984664becf46e1386
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/config.novacustom_v5*tnx: disable debug, bump edk2 for dgpu, bump rc

Change-Id: I2c59a11a1a7db508f605b6ebc965e34ffcfb4e88
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h: add dGPU ASL code

based on Google Brox / Hades boards, but heavily modified for our
purposes.

Change-Id: I05ad1edfe4bc9c88bd75c3d043edca4ded1dbaef
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec commented on src/mainboard/google/brox/acpi/power.asl:
```diff
@@ -199,39 +199,13 @@ Method (PGON, 0, Serialized)
 	/* Assert PERST# */
 	CTXS (GPIO_GPU_PERST_L)
 
-	/* Ramp up 1.2V rail on boards with support */
```
whoopsie

mkopec committed:
clevo/mtl-h/var/dgpu: add audio configuration

Change-Id: Ic3d6135c6373daeb49d6c718ae495a41939a4f1b
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/config.novacustom_v5.0tnx: bump edk2 for GOP fix + bump rc ver

Change-Id: Ib0267d07137cfe8827141dbd71afa5ba869265ab
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/var/dgpu: fixup audio config

Change-Id: I24eec94c7b77dc9a78e5f31e20c987a6d7e4d9b9
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/var/dgpu/data.vbt: enable fixed mode

Change-Id: I3b0f4e7bbc5f9dc485afa81b8668e51c61af74fe
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/config.novacustom_v5.0tnx: unhide HECI1

Change-Id: I7c63523ffe866961f845ee317added9a400ef12d
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/acpi/dgpu: make dynamic boost work

Change-Id: I8c31bb9472ce11396a8fed3594493cd7a888b596
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/config.novacustom_v5.0tnx: bump to rc4, set static edk2 rev, disable EDK2 serial by default

Change-Id: I803f68945e5fd5e394e43fbbbb61bee6a1bb08df
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

philipanda committed:
soc/intel/meteorlake: Hook the VT-d DMA protection option

Co-authored-by: Michał Żygowski <michal.zygowski@3mdeb.com>
Signed-off-by: Filip Gołaś <filip.golas@3mdeb.com>

miczyg1 committed:
Fix inconsistencies with schematics

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

mkopec committed:
soc/intel/mtl/romstage/romstage.c: fix typo and add missing include for VT-d

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/ramstage.c: skip disabling PTT

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/config.novacustom_v5.0tnx: bump to rc5

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

miczyg1 commented:
@mkopec something wrong happened, because the PR contains changes from iGPU variant

mkopec commented:
@miczyg1 rebased, should be okay now

miczyg1 reviewed (commented):
Not sure how useful these comments will be. Without a spec it is hard to say what should be done and how. There is a lot of other things in original firmware ACPI. Leaving it a sa comment rather than change request,

mkopec commented on src/mainboard/clevo/mtl-h/acpi/dgpu/gps.asl:
```diff
@@ -0,0 +1,89 @@
+		}
+		Case (GPS_FUNC_PSHAREPARAMS)
+		{
+			CreateField (Arg1, 0, 4, QUTY)	 /* Query type */
```
Original firmware has all functions in the GPU's _DSM method, here we have a separate method for each function and call GPS(Arg2, Arg3) in the _DSM

mkopec committed:
mb/clevo/mtl-h/acpi/dgpu: fixes after review

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec commented on src/mainboard/clevo/mtl-h/acpi/dgpu/nvjt.asl:
```diff
@@ -0,0 +1,129 @@
+				(0 << 12) |		/* NOT: Supports Notify on GC6 State done */
+				(1 << 13) |		/* MHYB: MS Hybrid Support (deferred GC6) */
+				(0 << 14) |		/* RPC: Root Port Control */
+				(0 << 15) |		/* GC6 Version (GC6-E) */
```
Changed to 0x2 as in original fw

mkopec commented on src/mainboard/clevo/mtl-h/acpi/dgpu/nvjt.asl:
```diff
@@ -0,0 +1,129 @@
+				(1 << JT_FUNC_SUPPORT) |
+				(1 << JT_FUNC_CAPS) |
+				(1 << JT_FUNC_POWERCONTROL) |
+				(1 << JT_FUNC_PLATPOLICY)))
```
yep, removed

mkopec commented on src/mainboard/clevo/mtl-h/acpi/dgpu/nvpcf.asl:
```diff
@@ -0,0 +1,101 @@
+				0x00,			/* [3:0] CPU type (0=Intel, 1=AMD),
+							   [7:4] GPU type (0=Nvidia) */
+
+				/* System Controller Table Header (v2.2), 1 controller entry */
```
Changed to 2.4 in https://github.com/Dasharo/coreboot/pull/548/commits/950f78bdbfaa995ac71ff9a140d9469c6d9f253f

mkopec commented on src/mainboard/clevo/mtl-h/acpi/dgpu/nvpcf.asl:
```diff
@@ -0,0 +1,101 @@
+
+			Local0 = Buffer (0x31) {
+				/* Dynamic Params Table Header (1 controller entry, 0x1c bytes) */
+				0x22, 0x05, 0x10, 0x1c, 0x01 }
```
Changed to 2.4 in https://github.com/Dasharo/coreboot/pull/548/commits/950f78bdbfaa995ac71ff9a140d9469c6d9f253f

mkopec commented on src/mainboard/clevo/mtl-h/acpi/dgpu/nvpcf.asl:
```diff
@@ -0,0 +1,101 @@
+							   [0:0] DC support
+							      0=Not supported, 1=Supported
+							   [31:1] Reserved. Set to 0. */
+				0x00, 0x00, 0x00, 0x00,
```
Changed to 0x1 in https://github.com/Dasharo/coreboot/pull/548/commits/950f78bdbfaa995ac71ff9a140d9469c6d9f253f

mkopec committed:
mb/clevo/mtl-h/acpi/dgpu: fixes to make it closer to clevo FW

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/novacustom_v5.0tnx: Enable UEFI GOP on nvidia dGPU

Enable option ROM loading and set resolution to 1920x1080 to maintain
readability on HiDPI screens.

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/novacustom_v5.0tnx: bump ver to -rc6

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/novacustom_v5.0tnx: bump edk2 for power failure string update

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/config.novacustom_v5.0tnx: disable opROM loading

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h: detect smartamp before enabling it

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec commented on src/mainboard/clevo/mtl-h/acpi/dgpu/gps.asl:
```diff
@@ -0,0 +1,89 @@
+
+/* GPS return Package */
+Name (GPSP, Buffer (0x28) {0x0})
+CreateDWordField (GPSP, 0, RETN)
+CreateDWordField (GPSP, 4, VRV1)
+CreateDWordField (GPSP, 8, TGPU)
```
Yes, but it doesn't actually populate most of them. Tested by calling the DSM by acpidbg, we should now be returning the same exact values

mkopec commented on src/mainboard/clevo/mtl-h/acpi/dgpu/nvop.asl:
```diff
@@ -0,0 +1,68 @@
+			Return (ITOB (
+				(1 << 0) |	/* Optimus Enabled */
+				(Local0 << 3) |	/* Current GPU Control Status */
+				(0 << 6) |	/* Shared Discrete GPU Hotplug Capabilities */
```
yes, fixed in https://github.com/Dasharo/coreboot/pull/548/commits/554b771a19486fb4979cdfc84fe57ac18600f291

mkopec committed:
configs/config.novacustom_v5.0tnx: enable prefmem alloc above 4G

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

miczyg1 committed:
mb/clevo/mtl-h/ramstage.c: Add port reset message for USB ports 2 and 6

These UPDs have to be set for PCH USB2.0 ports that are paired with CPU
XHCI controller.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

mkopec committed:
src/mainboard/clevo/mtl-h/Kconfig: remove S3

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/dt.cb: set C-state demotion to default

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/fadt.c: remove incorrect comment

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/inc/mainboard/variants.h: fix incorrect ifdef

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/ramstage.c: move comment to start of pinmux block

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/romstage.c: add Load Line config from Clevo FW

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/ramstage.c: disable S0i2.x substates

There is an issue with wake from deeper S0i2 substates on this hw.

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/config.novacustom_v5.0tnx: disable DMA protection option

Not functional with current FSP.

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/config.novacustom_v5.0tnx: bump to rc7

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
soc/intel/mtl/acpi/pcie.asl: fix RP12 IRQ mapping

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/config.novacustom_v5.0tnx: fix GOP driver path

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
ec/system76/ec/acpi/ec_ram.asl: add d-notifier ACPI field

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/acpi/dgpu/gpu_ec.asl: add back D-notifier support

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/acpi/dgpu/gps.asl: wire up D-notifier

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
ec/system76/ec/acpi: add DTT Power and Battery participants

Add DTT participants for power (with APWR and PSRC fields) and battery
devices.

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
mb/clevo/mtl-h/acpi/dgpu: remove NVPCF

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
soc/intel/tgl: add ACPI _HIDs for DPTF power and battery

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/config.novacustom_v5.0tnx: bump to rc8

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec commented on src/mainboard/clevo/mtl-h/acpi/dgpu/gpu_top.asl:
```diff
@@ -0,0 +1,98 @@
+		* Arg1 == force notification if no change (0 or 1)
+		*/
+		Name (CDNV, D1_EC)
+		Method (DNOT, 2, Serialized)
```
It's now used: https://github.com/Dasharo/coreboot/pull/548/commits/bab51b072f04e36df4db7cbeca4f09e4723dd1fd

mkopec committed:
configs/config.novacustom_v5.0tnx: bump EDK2 to fix missing subhook module

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/config.novacustom_v5.0tnx: remove rc suffix

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
.github/workflows/build.yml: build NCM MTL dGPU models too

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
.github/workflows/build.yml: temporarily disable EC building

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

mkopec committed:
configs/config.novacustom_n*: disable EC update

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

miczyg1 commented on .github/workflows/build.yml:
```diff
@@ -11,37 +11,16 @@ on:
       - '*'
 
 jobs:
-  build_novacustom_ec:
```
@mkopec I think we can still build EC regardless if EC sync is used or not. The "Obtain EC" step is removed and the dependency too, so it should be fine.

mkopec replied:
Right, commit amended: https://github.com/Dasharo/coreboot/commit/c2b15730096a54a4518709d9390e66fa8d58acea
